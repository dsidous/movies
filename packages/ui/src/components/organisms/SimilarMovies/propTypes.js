import { array, func } from 'prop-types';

export const propTypes = {
  similar: array.isRequired,
  handleMovieClick: func.isRequired,
};

export const defaultProps = {};
