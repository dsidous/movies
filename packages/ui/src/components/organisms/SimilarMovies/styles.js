import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      borderTop: theme.separator,
      margin: '40px auto 0',

      '& h4': {
        padding: '10px 0 20px',
      },
    },

    slider: {
      [theme.breakpoints.up('sm')]: {
        margin: '0 40px',
      },
    },

    slide: {
      cursor: 'pointer',
      textAlign: 'center',
      padding: '0 10px',

      '& img': {
        borderRadius: '4px',
        width: '100%',
      },

      '& small': {
        display: 'block',
        padding: 10,
      },
    },
  }),
  { name: 'similarShows' },
);
