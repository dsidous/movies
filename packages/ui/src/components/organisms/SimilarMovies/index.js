import React from 'react';
import Slider from 'react-slick';

import { propTypes } from './propTypes';
import MediaImage from '../../atoms/MediaImage';

const SimilarMovies = ({ similar, handleMovieClick }) => {
  const settings = {
    dots: false,
    infinite: true,
    speed: 500,
    slidesToShow: 7,
    slidesToScroll: 1,
  };

  if (window.innerWidth < 640) {
    Object.assign(settings, {
      arrows: false,
      slidesToScroll: 1,
      slidesToShow: 3,
      swipeToSlide: true,
    });
  }

  return (
    <div className="similar">
      <h4>Similar movies</h4>
      <Slider {...settings}>
        {similar.map(sim => (
          <div
            key={sim.id}
            className="similar-movies__element"
            onClick={() => handleMovieClick(sim.id)}
            onKeyDown={() => handleMovieClick(sim.id)}
            role="link"
            tabIndex="-1"
          >
            <p className="crew-col text-center">
              <MediaImage
                mediaType="poster"
                size={1}
                filePath={sim.poster_path}
                name={sim.title}
                className="similar-movies-img"
              />
              <br />
              <small>{sim.title}</small>
            </p>
          </div>
        ))}
      </Slider>
    </div>
  );
};

SimilarMovies.propTypes = propTypes;

export default SimilarMovies;
