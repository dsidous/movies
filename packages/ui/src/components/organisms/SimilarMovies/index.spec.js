import React from 'react';
import { shallow } from 'enzyme';
import SimilarMovies from '.';

const mockProps = {
  similar: [
    {
      id: 1,
      poster_path: 'path/to/image',
      title: 'title',
    },
  ],
  handleMovieClick: jest.fn(() => {}),
};

describe('Organisms/SimilarMovies', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<SimilarMovies {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle onClick event', () => {
    const wrapper = shallow(<SimilarMovies {...mockProps} />);

    wrapper.find('.similar-movies__element').first().simulate('click');
    expect(mockProps.handleMovieClick).toBeCalled();

    wrapper.find('.similar-movies__element').first().simulate('keyDown');
    expect(mockProps.handleMovieClick).toBeCalled();
  });
});
