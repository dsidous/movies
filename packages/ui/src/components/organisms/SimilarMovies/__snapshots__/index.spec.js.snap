// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/SimilarMovies should render as expected 1`] = `ShallowWrapper {}`;
